src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -341,11 +338,16 @@ class TypeErasure(isJava: Boolean, isSemi: Boolean, isConstructor: Boolean, wild
 
   private def eraseArray(tp: RefinedType)(implicit ctx: Context) = {
     val defn.ArrayType(elemtp) = tp
+    val arrayErasure = erasureFn(isJava, isSemi = false, isConstructor, wildcardOK)(_)
```

Comment:
You mean just replacing `val` by `def` in that line? Can you explain how that's better exactly?